src/scf/newton.jl

```diff
@@ -0,0 +1,410 @@
+# Newton's algorithm for the direct minimization of the energy
+# ------------------------------------------------------------
+#
+# Newton algorithm consist of iterating over density matrices like
+#       P = P - δP
+# where δP solves
+#       (Ω+K)δP = [P,[P,H(P)]] (residual)
+# where (Ω+K) if the Jacobian of the direct minimization iterations. It is
+# defined as a super operator from the space tangent to the constraints
+# manifold at point P_∞, the solution of our problem.
+#   - K is the Hessian of the energy on the manifold;
+#   - Ω represents the influence of the curvature of the manifold :
+#         ΩδP = -[P_∞,[H(P_∞),δP]].
```

Comment:
Could mention for stupid folks like me that its the negative inverse of the occupation function.